### DISCUSSION TITLE
[UDF] How to load udf jar in pyhive ?

### DISCUSSION THREAD

**Discussion by BruceWong96** [apache/kyuubi#4971] [Q&A] [answered]
In kyuubi doc, the engine side can finish loading udf .

<img width="795" alt="image" src="https://github.com/apache/kyuubi/assets/29974394/bccc2201-e877-482a-863c-d4baee45d543">

So, how to do it. My test code is below.

    configuration = get_set_command(sql_0)
    # dev
    # kyuubi_host = 'x.x.x.x'
    # prod
    kyuubi_host = 'x.x.x.x'

    cursor = hive.connect(host=kyuubi_host, port=8888, username='user01', configuration=configuration).cursor()

    sql_arr = sql_0.split(u';')
    for index in range(len(sql_arr)):
        sql_arr[index] = sql_arr[index]

    # for sql in sql_arr:
        # print(sql)
        # print('---------++++++++++++++-------------')

    for sql in sql_arr:
        cursor.execute(sql)
        print('---------++++++++++++++-------------')
        print(cursor.fetchall())`

Thanks.

**Comment by pan3793** [comment, marked as answer]
Call `ADD JAR ...` using SQL `cursor.execute(sql)`

https://spark.apache.org/docs/latest/sql-ref-syntax-aux-resource-mgmt-add-jar.html

**Reply by BruceWong96** [reply]
Thanks
